File: ru.kmorozov.library.data.client/src/main/java/ru/kmorozov/library/data/client/netty/ServerAdapterInitializer.java
package ru.kmorozov.library.data.client.netty;

import io.netty.channel.ChannelInitializer;
import io.netty.channel.ChannelPipeline;
import io.netty.channel.socket.SocketChannel;
import io.netty.handler.codec.string.StringDecoder;
import io.netty.handler.codec.string.StringEncoder;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class ServerAdapterInitializer extends ChannelInitializer<SocketChannel> {

    @Autowired
    private ServerAdapterHandler handler;

    @Override
    protected void initChannel(SocketChannel channel) throws Exception {
        ChannelPipeline pipeline = channel.pipeline();

        pipeline.addLast("decoder", new StringDecoder());
        pipeline.addLast("encoder", new StringEncoder());

        pipeline.addLast("handler", handler);
    }

}

File: ru.kmorozov.library.data.client/src/main/java/ru/kmorozov/library/data/client/netty/ServerAdapterInitializer.kt
package ru.kmorozov.library.data.client.netty

import io.netty.channel.ChannelInitializer
import io.netty.channel.ChannelPipeline
import io.netty.channel.socket.SocketChannel
import io.netty.handler.codec.string.StringDecoder
import io.netty.handler.codec.string.StringEncoder
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class ServerAdapterInitializer : ChannelInitializer<SocketChannel>() {

    @Autowired
    private val handler: ServerAdapterHandler? = null

    override fun initChannel(channel: SocketChannel) {
        val pipeline = channel.pipeline()

        pipeline.addLast("decoder", StringDecoder())
        pipeline.addLast("encoder", StringEncoder())

        pipeline.addLast("handler", handler)
    }

}
